Library/Homebrew/utils/shared_audits.rb

```diff
@@ -92,6 +92,48 @@ def gitlab_release(user, repo, tag, formula: nil, cask: nil)
     "#{tag} is a GitLab pre-release."
   end
 
+  def codeberg_repo_data(user, repo)
+    @codeberg_repo_data ||= {}
+    @codeberg_repo_data["#{user}/#{repo}"] ||= begin
+      out, _, status = Utils::Curl.curl_output("https://codeberg.org/api/v1/repos/#{user}/#{repo}")
+      json = JSON.parse(out) if status.success?
+      json = nil if json&.dig("message")&.include?("404 Project Not Found")
+      json
+    end
+  end
+
+  def codeberg_release_data(user, repo, tag)
+    id = "#{user}/#{repo}/#{tag}"
+    @codeberg_release_data ||= {}
+    @codeberg_release_data[id] ||= begin
+      out, _, status = Utils::Curl.curl_output(
+        "https://codeberg.org/api/v1/repos/#{user}/#{repo}/releases/tags/#{tag}", "--fail"
+      )
+      JSON.parse(out) if status.success?
+    end
+  end
+
+  def codeberg_release(user, repo, tag, formula: nil, cask: nil)
+    release = codeberg_release_data(user, repo, tag)
+    return unless release
+
+    return if DateTime.parse(release["released_at"]) <= DateTime.now
+
+    exception, name, version = if formula
+      [formula.tap&.audit_exception(:codeberg_prerelease_allowlist, formula.name), formula.name, formula.version]
+    elsif cask
+      [cask.tap&.audit_exception(:codeberg_prerelease_allowlist, cask.token), cask.token, cask.version]
```

Comment:
Do we need this? Can we hold off until we do?

Reply:
sure, we can remove that for now (but codeberg's api is very similar to github though)